lib/internal/webstreams/writablestream.js

```diff
@@ -157,21 +157,31 @@ class WritableStream {
   [kType] = 'WritableStream';
 
   /**
-   * @param {UnderlyingSink} [sink]
+   * @param {UnderlyingSink} sink
    * @param {QueuingStrategy} [strategy]
    */
-  constructor(sink = kEmptyObject, strategy = kEmptyObject) {
+  constructor(sink, strategy = kEmptyObject) {
+    if (!sink) throw new ERR_INVALID_ARG_VALUE('sink', 'WritableStream sink is required');
     markTransferMode(this, false, true);
     validateObject(sink, 'sink', kValidateObjectAllowObjects);
     validateObject(strategy, 'strategy', kValidateObjectAllowObjectsAndNull);
     const type = sink?.type;
     if (type !== undefined)
       throw new ERR_INVALID_ARG_VALUE.RangeError('type', type);
 
+    if (typeof sink.write !== 'function') {
+      throw new ERR_INVALID_ARG_VALUE.TypeError(
+        'sink',
+        `WritableStream sink must implement a write() method. ` +
+        `If you want a writable stream that silently consumes and discards data, ` +
+        `consider using DummyWritableStream instead.`
+      );
+    }
```

Comment:
This is not required either - https://streams.spec.whatwg.org/#underlying-sink-api 

"Such objects *can* contain any of the following properties..."